includes/class-guest-session.php

```diff
@@ -46,6 +46,22 @@ public static function get_current_guest() {
 		return $session->user;
 	}
 
+	/**
+	 * Checks if the current guest user has an account.
+	 *
+	 * @return bool
+	 */
+	public static function current_guest_has_account() : bool {
+		$session = self::instance();
+		$session->init();
+
+		if ( false === $session->user ) {
+			return false;
+		}
+
+		return (bool) get_user_by( 'email', $session->user->user_email );
```

Comment:
Just a suggestion to improve a bit the code, increasing code re-use. :) 

```suggestion
		$user = self::get_current_guest();

		if ( false === $user ) {
			return false;
		}

		return (bool) get_user_by( 'email', $user->user_email );
```